### PR TITLE
[th/host-cancellable] make Host.run() cancellable and add common.{FutureThread,Cancellable}

### PR DESCRIPTION
**Pull request by thom311**
This allows to run a background process, and cancel (kill) it.

### CODE REVIEW

**Comment by wizhaoredhat** [comment]
This is a very good PR and will help with cleaning up forked processes. I think someone else on the team could also give this review!

File: ktoolbox/common.py [hunk 1]
```diff
@@ -1533,3 +1534,59 @@ def get_poll_fd(self) -> int:
...
+        self._user_data: Any = None
+        if start:
+            self.start()
+
+    @property
+    def cancellable(self) -> Cancellable:
+        with self._lock:
+            if self._cancellable is None:
+                self._cancellable = Cancellable()
+                if self._cancellable_is_cancelled:
```
Comment (SalDaniele) [inline ktoolbox/common.py:1562]: Why not check self._cancellable.is_cancelled?

Comment (thom311) [reply ktoolbox/common.py:1562]: the `FutureThread`'s `self._cancellable` is only created if a caller requests it. This probably could be simplified by letting `FutureThread.__init__()` always create a cancellable, but the way it is, it's optimized to only create it once it's used.

Just look at the 3 uses of `_cancellable_is_cancelled`. As we create the cancellable only when requested, it is necessary to remember if the cancellable was cancelled (even if the cancellable does not yet exist). This is also covered by a unit test.

**Comment by SalDaniele** [comment]
LGTM

**Pull request merged**